solidity/dashboard/src/components/TokenAmount.jsx

```diff
@@ -12,14 +12,21 @@ const TokenAmount = ({
   displayWithMetricSuffix,
   currencySymbol,
   displayAmountFunction,
+  withTooltip,
+  tooltipText,
 }) => {
   const { value, suffix } = displayWithMetricSuffix
     ? getNumberWithMetricSuffix(displayAmountFunction(amount, false))
```

Comment:
The more I think about it the more it makes sense to simplify this and drop `displayAmountFunction`, `withTooltip`, and `tooltipText` entirely, and do all of the formatting internally.

To do this, we might rename `displayEthAmount` to `display18DecimalAmount` or similar. `TokenAmount` would instead allow passing in e.g. `smallestPrecisionDecimal` and `smallestPrecisionUnit`. For ETH, `currencySymbol` might be `ETH`, `14`, and `gwei`. TBTC might use this as `8` and `tSats`, for example, to display `<0.0001 TBTC` and a tooltip of `1000.01256 tSats`.

Thoughts? I think it can happen in a followup, of course.

Reply:
I think it makes sense. I'll open an issue for that, since it won't be a one-line fix.

Reply:
#2095